--- gnugo-3.8.orig/interface/play_ascii.c
+++ gnugo-3.8/interface/play_ascii.c
@@ -950,8 +950,10 @@
 	  strtok(command, " ");
 	  tmpstring = strtok(NULL, " ");
 	  if (tmpstring) {
+#if !READLINE
 	    /* discard newline */
 	    tmpstring[strlen(tmpstring) - 1] = 0;
+#endif
 	    /* make sure we are saving proper handicap */
 	    init_sgf(gameinfo);
 	    writesgf(sgftree.root, tmpstring);
@@ -967,8 +969,10 @@
 	  strtok(command, " ");
 	  tmpstring = strtok(NULL, " ");
 	  if (tmpstring) {
+#if !READLINE
 	    /* discard newline */
 	    tmpstring[strlen(tmpstring) - 1] = 0;
+#endif
 	    if (!sgftree_readfile(&sgftree, tmpstring)) {
 	      fprintf(stderr, "Cannot open or parse '%s'\n", tmpstring);
 	      break;
@@ -1069,8 +1073,10 @@
       strtok(command, " ");
       tmpstring = strtok(NULL, " ");
       if (tmpstring) {
+#if !READLINE
 	/* discard newline */
 	tmpstring[strlen(tmpstring) - 1] = 0;
+#endif
 	init_sgf(gameinfo);
 	writesgf(sgftree.root, tmpstring);
       }
